src/common/copilot/assets/scripts/copilot.js

```diff
@@ -487,13 +461,21 @@
     }
         case "selectedCodeInfo": {
             const chatInputLabel = document.getElementById("input-label-id");
-            if (message.value.start == message.value.end && message.value.selectedCode.length == 0) {
+            selectedCode = message.value.selectedCode;
+            if (message.value.start == message.value.end && selectedCode.length == 0) {
                 chatInputLabel.classList.add("hide");
                 break;
             }
             chatInputLabel.classList.remove("hide");
             chatInputLabel.innerText = `Lines: ${message.value.start + 1} - ${message.value.end + 1} selected`;
+            break;
+        }
+        case "explainCode": {
+            selectedCode = message.value.selectedCode;
+            const explainPrompt = 'Explain the following code:';
```

Comment:
String needs to be localized

Reply:
I'm not sure if the `.js` asset files can be, or perhaps just not easily with the existing vscode pseudo-module.

IIRC, Nidhi mentioned that copilot isn't yet available outside of US region and en-US language, so this might not need localization yet?

Reply:
That's correct, current vscode localization module is not supported in webviews and currently copilot is only available in US region.

Reply:
Even if it is not supported right now. It might be good idea to move strings to one location so localization later can be easily done

Reply:
Right, will be moving all the string out of webview. So, they can be localized first and then sent to webview. Will take this up in separate as it will changes for all existing strings also.

Reply:
It would be great to have it in one PR

Reply:
Added